cpp/velox/compute/VeloxPlanConverter.cc

```diff
@@ -34,183 +33,87 @@ VeloxPlanConverter::VeloxPlanConverter(
     const std::unordered_map<std::string, std::string>& confMap,
     const std::optional<std::string> writeFilesTempPath,
     bool validationMode)
-    : inputIters_(inputIters),
-      validationMode_(validationMode),
+    : validationMode_(validationMode),
       substraitVeloxPlanConverter_(veloxPool, confMap, writeFilesTempPath, validationMode),
-      pool_(veloxPool) {}
-
-void VeloxPlanConverter::setInputPlanNode(const ::substrait::WriteRel& writeRel) {
-  if (writeRel.has_input()) {
-    setInputPlanNode(writeRel.input());
-  } else {
-    throw std::runtime_error("Child expected");
-  }
-}
-
-void VeloxPlanConverter::setInputPlanNode(const ::substrait::FetchRel& fetchRel) {
-  if (fetchRel.has_input()) {
-    setInputPlanNode(fetchRel.input());
-  } else {
-    throw std::runtime_error("Child expected");
-  }
-}
-
-void VeloxPlanConverter::setInputPlanNode(const ::substrait::ExpandRel& sexpand) {
-  if (sexpand.has_input()) {
-    setInputPlanNode(sexpand.input());
-  } else {
-    throw std::runtime_error("Child expected");
-  }
-}
-
-void VeloxPlanConverter::setInputPlanNode(const ::substrait::GenerateRel& sGenerate) {
-  if (sGenerate.has_input()) {
-    setInputPlanNode(sGenerate.input());
-  } else {
-    throw std::runtime_error("Child expected");
-  }
+      pool_(veloxPool) {
+  // avoid include RowVectorStream.h in SubstraitToVeloxPlan.cpp, it may cause redefintion of array abi.h.
+  auto factory = [](std::string nodeId,
+                    memory::MemoryPool* pool,
+                    std::shared_ptr<ResultIterator> inputIter,
+                    RowTypePtr outputType) {
+    auto vectorStream = std::make_shared<RowVectorStream>(pool, std::move(inputIter), outputType);
+    return std::make_shared<ValueStreamNode>(nodeId, outputType, std::move(vectorStream));
+  };
+  substraitVeloxPlanConverter_.setValueStreamNodeFactory(std::move(factory));
+  substraitVeloxPlanConverter_.setInputIterators(inputIters);
```

Comment:
If using this way, should we remove `setInputIterators` then capture `inputIters` in factory body?

Reply:
Sorry, I still don't quite understand the point here. Capturing `inputIters` won't make the factory stateful since once it is created, nobody could change its internal creation logic of the value stream node. And e could even eliminate stateful `setInputIterators` by the change as well. Any special reason you wanted to keep it?

Reply:
got it, thanks!